fix(h1): ignore chunked trailers

alpire opened the pull request:
Previously, hyper returned an "Invalid chunk end CR" error on chunked
responses with trailers, as described in RFC 7230 Section 4.1.2. This
commit adds code to ignore the trailers.

Closes #2171

seanmonstar reviewed (commented):
Thanks so much for writing this up! I really appreciate you including a test too!

For completeness, would you mind adding a couple more tests (either in `tests/`, or just testing the decoder in `proto::h1::decode::tests`)? It'd be good to check that this will handle multiple trailer liners (it looks like it does correctly), and perhaps some error cases.

alpire commented:
Thanks for the review and feedback! I just added the additional tests you mentioned:
- A test with multiple trailers (in `tests/`)
- A test checking the new error case I introduced, occurring when the trailer is missing a new line (in `proto::h1::decode::tests`)

seanmonstar reviewed (approved):
Awesome work, thanks!
